### PR TITLE
Kirill.sitcevoy/box drawing

### PR DESCRIPTION
**Pull request by kievzenit**
Box drawing home task

### CODE REVIEW

File: DrawingBox/DrawingBox/Box.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                Console.Write(Box.borderSymbol);
+            }
+
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+        private static void drawMessage()
+        {
+            Console.ForegroundColor = ConsoleColor.Yellow;
+
+            var messageArea = (Box.height - 2) * (Box.width - 2);
```
Comment (SergiySimchuk) [inline DrawingBox/DrawingBox/Box.cs]: (Box.height - 2) and (Box.width - 2)
You repeat these calculations couple times. I think better performance way to declare 2 new variables, calculate these values once and use they during executing methods.

File: DrawingBox/DrawingBox/Box.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            if (Box.message.Length > messageArea)
+            {
+                Box.message = Box.message.Substring(0, messageArea - 3);
+                Box.message += "...";
+            }
+
+            var topOffset = 0;
+            var leftOffset = 0;
+
+            var rows = Box.message.Length / (Box.width - 2);
```
Comment (SergiySimchuk) [inline DrawingBox/DrawingBox/Box.cs]: What if "var rows =" will receive double type?

**Review by SergiySimchuk — APPROVED** [review]

File: DrawingBox/DrawingBox/Box.cs (hunk 2 above)
Comment (kievzenit) [reply DrawingBox/DrawingBox/Box.cs]: It will always be int, left and right operand is int

File: DrawingBox/DrawingBox/Box.cs (hunk 1 above)
Comment (kievzenit) [reply DrawingBox/DrawingBox/Box.cs]: Fixed in commit 3246ffc9de4f9d210989bb2bd682c40eebcd0e94

File: DrawingBox/DrawingBox/Box.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+            Box.drawBorder();
+            Box.drawMessage();
+        }
+        private static void drawBorder()
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+
+            Console.SetCursorPosition(Box.x, Box.y);
+            Console.WriteLine(new string(Box.borderSymbol, Box.width));
```
Comment (horianyn) [inline DrawingBox/DrawingBox/Box.cs:50]: Nice choise using "new string"

File: DrawingBox/DrawingBox/Box.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+            var rows = Box.message.Length / (Box.width - 2);
+            if (Box.height - 2 != rows)
+                topOffset = ((Box.height - rows) / 2) - 1;
+
+            if (Box.message.Length < Box.width - 2)
+                leftOffset = ((Box.width - 2) - Box.message.Length) / 2;
+
+            var counter = 0;
+            for (int yPosition = 1; yPosition <= messageArea / (Box.width - 2); yPosition++)
```
Comment (horianyn) [inline DrawingBox/DrawingBox/Box.cs]: You could use (Box.height - 2). It is the same as: `messageArea / (Box.width - 2)`

File: DrawingBox/DrawingBox/Box.cs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            {
+                for (int xPosition = 1; xPosition <= messageArea / (Box.height - 2); xPosition++)
```
Comment (horianyn) [inline DrawingBox/DrawingBox/Box.cs]: You could use (Box.width - 2). It is the same as: messageArea / (Box.height - 2)

**Review by horianyn — APPROVED** [review]

File: DrawingBox/DrawingBox/Box.cs (hunk 3 above)
Comment (kievzenit) [reply DrawingBox/DrawingBox/Box.cs:50]: Thanks

File: DrawingBox/DrawingBox/Box.cs (hunk 4 above)
Comment (kievzenit) [reply DrawingBox/DrawingBox/Box.cs]: Agree with you, fixed in commit 501ea96c2452991df69ed1f139a02b7f8242edb1

File: DrawingBox/DrawingBox/Box.cs (hunk 5 above)
Comment (kievzenit) [reply DrawingBox/DrawingBox/Box.cs]: Fixed in commit 501ea96c2452991df69ed1f139a02b7f8242edb1

File: DrawingBox/DrawingBox/Box.cs [hunk 6]
```diff
@@ -0,0 +1,137 @@
+﻿using System;
+
+namespace DrawingBox
+{
+    public static class Box
```
Comment (Artya) [inline DrawingBox/DrawingBox/Box.cs:5]: why this class is static? do this has any sense?

Comment (kievzenit) [reply DrawingBox/DrawingBox/Box.cs:5]: Yeah, it does. This class just draws a message you enter in. Message that is printed out holdes the whole console, and before another message this class clears console, I don't think that someone needs a couple of instance of this class.

**Pull request closed without merging**